### ISSUE TITLE
[Bug]: Querynode keeps high cpu utilization, hours after our last query

### ISSUE THREAD

**Issue by parsa-ra** [milvus-io/milvus#11053]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.0.0rc7
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): v2.0.0rc8
- OS(Ubuntu or CentOS): Ubuntu 20.04.3 LTS
- CPU/Memory: 16 Core / 48 GB
- GPU: None 
- Others:
```

### Current Behavior

Incremental insertion of about 2 million vectors and creation of index (IVF-SQ8, nlist=16384), successfully can make queries but the CPU usage of querynode is extremely high even though its current stage (which no query currently passes to the server) is extremely high.

Check out the following image:
![image](https://user-images.githubusercontent.com/11356471/139731328-1bbf02fc-b67d-4347-819c-b0bb7837793d.png)

Here is the output of the `kubectl get pods`:
```bash
NAME                                       READY   STATUS      RESTARTS   AGE
mil-etcd-0                                 1/1     Running     0          33h
mil-etcd-1                                 1/1     Running     0          33h
mil-etcd-2                                 1/1     Running     0          33h
mil-milvus-datacoord-5557997cd7-64qdk      1/1     Running     0          33h
mil-milvus-datanode-7b64f7b79d-9z4rr       1/1     Running     0          33h
mil-milvus-indexcoord-6684895cb4-xrlh9     1/1     Running     0          33h
mil-milvus-indexnode-b86d5fb85-bsmt2       1/1     Running     0          33h
mil-milvus-indexnode-b86d5fb85-p4tbz       1/1     Running     0          33h
mil-milvus-proxy-54769646b4-f9b84          1/1     Running     0          33h
mil-milvus-proxy-54769646b4-tzfft          1/1     Running     0          33h
mil-milvus-querycoord-f98c4ddd5-k48vb      1/1     Running     0          33h
mil-milvus-querynode-6bc658d7-6n46n        1/1     Running     0          33h
mil-milvus-querynode-6bc658d7-z9mqm        1/1     Running     0          33h
mil-milvus-rootcoord-99b69b8f5-2fcz4       1/1     Running     0          33h
mil-minio-0                                1/1     Running     0          33h
mil-minio-1                                1/1     Running     0          33h
mil-minio-2                                1/1     Running     0          33h
mil-minio-3                                1/1     Running     0          33h
mil-pulsar-autorecovery-79c9d7d4c8-bpkk4   1/1     Running     0          33h
mil-pulsar-bastion-f54cbd96d-h67n2         1/1     Running     0          33h
mil-pulsar-bookkeeper-0                    1/1     Running     0          33h
mil-pulsar-bookkeeper-1                    1/1     Running     0          33h
mil-pulsar-broker-7d676ddcc8-c2gb8         1/1     Running     0          33h
mil-pulsar-proxy-565b8bdbc4-m7ncs          2/2     Running     0          33h
mil-pulsar-zookeeper-0                     1/1     Running     0          33h
mil-pulsar-zookeeper-1                     1/1     Running     0          33h
mil-pulsar-zookeeper-2                     1/1     Running     0          33h
mil-pulsar-zookeeper-metadata--1-mnlcn     0/1     Completed   0          33h
```

And here is the last few logs of the querynodes in order:
[querynode1.txt](https://github.com/milvus-io/milvus/files/7455739/querynode1.txt)
[querynode2.txt](https://github.com/milvus-io/milvus/files/7455740/querynode2.txt)

**Comment by yanliang567** [comment]
@parsa-ra  did you load the collection before CPU  utilization goes high?

/assign @xige-16 
Could you please help to take a look at this issue?

/unassign

**Comment by parsa-ra** [comment]
Yeah, I believe it is not possible to perform a search without loading the collections first, am I wrong? 
![image](https://user-images.githubusercontent.com/11356471/139784356-40bd0b89-0904-4437-87b4-ccc0ad636063.png)

**Comment by xiaofan-luan** [comment]
@bigsheeper  please take a look on this issue?

**Comment by bigsheeper** [comment]
ok，I'm working on it

**Comment by bigsheeper** [comment]
Hi ~ @parsa-ra 
I'v tried to reproduce it on my computer(12 Core / 32 GB), I inserted 1 million vectors and also created index (IVF-SQ8, nlist=16384), it searched well and successfully. However, the CPU usage of querynode is quite normal after search or hours after search as shown below:

![Screenshot from 2021-11-18 20-27-17](https://user-images.githubusercontent.com/42060877/142416712-ce851635-5862-48fd-a627-992cd0f8bd8f.png)

![Screenshot from 2021-11-18 20-27-44](https://user-images.githubusercontent.com/42060877/142416768-42af5040-be58-4cb1-9949-6ab7b7f55c3f.png)


At your convenience, could you please provide me the full client codes and retest it again? Let me know if you need any help. 
Have a wonderful day!

**Comment by bigsheeper** [comment]
updata. inserted 2 million vectors and created index (IVF-SQ8, nlist=16384)

![Screenshot from 2021-11-19 09-53-30](https://user-images.githubusercontent.com/42060877/142552291-55688880-53c2-4d25-ae11-b5672fce1d89.png)


![Screenshot from 2021-11-19 09-53-45](https://user-images.githubusercontent.com/42060877/142552300-940f482a-470e-4796-b7b3-6e0e8b2af922.png)

**Comment by parsa-ra** [comment]
Are you using the same build as our test?
Strange!! I wonder are if all of your input data are sealed?

**Comment by bigsheeper** [comment]
Hello @parsa-ra , 
Thank you for your reply and apologies for my fault to use the `2.0.0-rc8` version of milvus to test it a few days ago. And I'v reproduce it on the `2.0.0-rc7` version.
There is a little issue with the `2.0.0-rc7` version to lead to high cpu usage in query node (#10722), and this issue has been fixed in the `2.0.0-rc8` version. So we recommend that you use the `2.0.0-rc8` version.
Hope that clarifies.

**Comment by parsa-ra** [comment]
Got it, thanks ...
